doc/sphinx/language/core/records.rst

```diff
@@ -177,6 +183,27 @@ Defining record types
          Class MyClass := { myfield2 : nat }.
          About myfield2. (* Argument name defaults to the class name and is marked implicit *)
 
+.. _record_where_clause:
+
+   .. example:: Using a :g:`where` clause in a record field
+
+      .. rocqtop:: all
+
+         Reserved Notation "a & b" (at level 40, left associativity).
+         Record nat_comoid :=
+         {
+           op : nat -> nat -> nat where "a & b" := (op a b);
+           identity : nat;
+           identity_cond : forall n, identity & n = n;
+           comm: forall a b, a & b = b & a;
+           assoc: forall a b c, a & (b & c) = a & b & c
+         }.
+
+   .. exn:: Error: "where" clause not supported for records.
+
+      :g:`where` clauses are only supported for :n:`@record_field`\s, not for the overall
+      :cmd:`Record` definition.
```

Comment:
This error message can appear for Record, Structure, Class, Inductive and CoInductive, so not quite right to refer to the Record command here.  But they are all definitions.

```suggestion
      :g:`where` clauses are only supported for :n:`@record_field`\s, not for the overall
      definition.
```

FWIW, the message wording `Error: "where" clause not supported for records.` is not great--Inductive and CoInductive are not records at all.  Class is a special case of a record.  Not sure about Structure--the doc says it's a synonym for Record, but a superficial examination of the code suggests that it's processed differently from Record--do either of you know how it differs?

Reply:
I think we are diverging from the original intent of this PR.
I guess records are not that much about the keyword defining them but about the syntax `record_definition` which makes them records.

Reply:
> I think we are diverging from the original intent of this PR.

How so?  You wanted to add the exception.  Immediately above, I'm suggesting a tiny correction to what you currently have for that.  And [here](https://github.com/rocq-prover/rocq/pull/20753#discussion_r2150846923) asks you to add your text in a different spot.  Everything else has been covered.